### ISSUE TITLE
[Question] How to use two fixture in one test suite file ?

### ISSUE THREAD

**Issue by amrsa1** [microsoft/playwright#12335]
i have created a fixture for multiple page object classes, is there way to import multiple fixture in on test suite ?

```
import { expect, Page } from '@playwright/test';
import { Search as test } from '../../pages/fixture';


test.describe('Search page Test Suite', () => {

    test.afterEach(async ({page}) => {
        page.close()
    });

    test('test example', async ({ page, browserName, publicSearch }) => {
        await page.goto('url', { waitUntil: "load" })
        expect(expected).toMatchSnapshot(`actual`);
    });
```

how i can add another fixture however im importing `Search as test`

**Comment by mxschmitt** [comment]
Yes, just define your other fixtures inside the object of your test. For example:

`test('test example', async ({ page, browserName, publicSearch, foobar }) => {`

**Comment by amrsa1** [comment]
@mxschmitt  im a bit confused since test is wrapped with the fixture as we are importing  fixture as test
`import { publicSearch as test } from '../../pages/fixture';`
so how i will add another fixture

**Comment by mxschmitt** [comment]
a fixture is a either a worker or a test fixture which you define with .extend see here: https://playwright.dev/docs/test-fixtures

If you have defined it inside your pages/fixtures then you can just use it inside your test.

Also I'd recommend to remove the test.afterEach, since we do close the browser for you already and you did not await the page.close()

**Comment by amrsa1** [comment]
@mxschmitt regarding page.close()

i have noticed that im running test against different browsers in same instance the browsers are not closed until all test suite is done which somehow consuming memory resources, i have noticed if i explicitly add it it will close the browser after each test not waiting until all are done

**Comment by yury-s** [comment]
> @mxschmitt im a bit confused since test is wrapped with the fixture as we are importing fixture as test
> `import { publicSearch as test } from '../../pages/fixture';`
> so how i will add another fixture

As long as fixtures declared in the same `test.extend` all of them will be available as the test parameters. Do you have fixtures declared in separate files? Can you show the code the with the fixture declarations and their intended usages so that it's easier for us to give you an advise?

> i have noticed that im running test against different browsers in same instance the browsers are not closed until all test suite is done which somehow consuming memory resources, i have noticed if i explicitly add it it will close the browser after each test not waiting until all are done

Normally the browser stays alive while the test worker is running and it is shared by multiple tests (by default all tests in a file would reuse same browser instance). The page and the context though are closed after each test, so closing it manually should have no effect and closing a page will not close its browser.
